app/uk/gov/hmrc/cataloguefrontend/whatsrunningwhere/WhatsRunningWhereController.scala

```diff
@@ -61,25 +62,13 @@ class WhatsRunningWhereController @Inject() (
         form                 <- Future.successful(WhatsRunningWhereFilter.form.bindFromRequest)
         profile               = profileFrom(form)
         selectedProfileType   = form.fold(_ => None, _.profileType).getOrElse(ProfileType.Team)
+        selectedViewMode      = form.fold(_ => None, _.viewMode).getOrElse(ViewMode.Versions)
         (releases, profiles) <- (service.releasesForProfile(profile).map(_.sortBy(_.applicationName.asString)), service.profiles).mapN((r, p) => (r, p))
         environments          = distinctEnvironments(releases)
+        serviceDeployments: Seq[ServiceDeploymentConfigSummary] <- service.allReleases(releases)
         profileNames          = profiles.filter(_.profileType == selectedProfileType).map(_.profileName).sorted
-      } yield Ok(page(environments, releases, selectedProfileType, profileNames, form, showDiff))
+      } yield Ok(page(environments, releases, selectedProfileType, profileNames, form, showDiff, serviceDeployments.sortBy(_.serviceName), config.maxMemoryAmount, selectedViewMode))
```

Comment:
wondering if the two controller approach might have been better here? you could avoid loading data you dont need to render a given view. If you wanted to keep a single endpoint/url you could break each view (including loading all its data) into its own function and have the endpoint call either one depending on the viewmode param?

Reply:
Is it not just the `serviceDeployments`?  You could just wrap L68 with an `if else`

Reply:
i think you might need to do something about releasesForProfile when not looking at releases (theres a connector to get just profiles)

Reply:
Have split created 2 functions that are called by the controller, meaning only the data we need is loaded :)